Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -2765,6 +2766,14 @@ public void onActivityStarted(Activity activity) {
                 initSessionWithData(intentData, activity); // indicate  starting of session.
             }
             activityCnt_++;
+
+            // Activity stack for finding the structured path to this activity
+            if (activityStack_.size() > 1 && (activityStack_.get(activityStack_.size() - 2)).equals(activity.getClass().getSimpleName())) {
+                activityStack_.remove(activityStack_.get(activityStack_.size() - 2));
+                activityStack_.remove(activityStack_.get(activityStack_.size() - 1));
```

Comment:
Since this is a System UI event it is always called in only in UI thread. This method is adopted for finding the the path to a current Activity (e.g.`home/recipe-list/recipe-12345`). Since we are not using this now we can remove this